### PR TITLE
Second attempt

### PR DESCRIPTION
**Pull request by brl0**
This pull request addresses issues mentioned in #98 and #99, which I opened and have since closed, and bumps the version in #97 as well.
In summary, this PR:
- prevents an error listing loaded objects with no __name__ attribute;
- corrects situation in which context is not handed off properly (when run with python -m);
- adds python headers to the .konchrc file and defaults for some editors to recognize syntax;
- uses importlib rather than deprecated imp module;
- has a couple of trivial cleanups.
As mentioned in my previous attempt, this is my first pull request, so my apologies for any missteps, and I would appreciate any appropriate guidance.
Thanks for the cool tool.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -46,3 +46,9 @@ README.html
 .mypy_cache
 .cache
 .konchrc.local
+
+# Visual Studio Code
+.vscode/
```
Comment (sloria) [inline .gitignore]: This belongs in your [global gitignore file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

File: .konchrc [hunk 2]
```diff
@@ -1,22 +1,19 @@
+#!/opt/conda/bin/python
```
Comment (sloria) [inline .konchrc]: `.konchrc` should not have shebang lines, since they aren't executed as scripts. Please remove this.

File: .konchrc [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
+# -*- coding: utf-8 -*-
```
Comment (sloria) [inline .konchrc]: The encoding pragma is unnecessary as of Python 3, and konch no longer supports Python 2. This should be removed.

File: .konchrc [hunk 4]
```diff
@@ -1,22 +1,19 @@
...
     {
-        "context": [
-            konch,
-            env,
-            Path,
-            IPython,
-            bpython,
-            ptpython,
-        ],
+        "context": [sys, os, konch, env, Path, IPython, bpython, ptpython],
```
Comment (sloria) [inline .konchrc]: This change (and the associated imports) is unrelated the fix, so please don't include them in the PR.

File: .konchrc [hunk 5]
```diff
@@ -27,3 +24,11 @@ konch.config(
 
 if Path(".konchrc.local").exists():
     konch.use_file(".konchrc.local", trust=True)
+
+
+def setup():
```
Comment (sloria) [inline .konchrc]: Remove these functions, as they're unrelated to the fix.

File: CHANGELOG.rst [hunk 6]
```diff
@@ -2,6 +2,14 @@
 Changelog
 *********
 
+4.2.2 (2019-06-06)
```
Comment (sloria) [inline CHANGELOG.rst]:
```suggestion
4.2.2 (unreleased)
```

I'll add the release date myself.

File: konch.py [hunk 7]
```diff
@@ -1005,7 +1016,9 @@ def edit_file(
             authfile.allow(filename)
 
 
-INIT_TEMPLATE = """# vi: set ft=python :
+INIT_TEMPLATE = """#!/opt/conda/bin/python
```
Comment (sloria) [inline konch.py]: See previous comment; the shebang and encoding pragma do not belong here.

File: setup.py [hunk 8]
```diff
@@ -8,7 +8,7 @@
         "mypy==0.701",
         "flake8==3.7.7",
         "flake8-bugbear==19.3.0",
-        "pre-commit==1.16.1",
+        "pre-commit==1.17.0",
```
Comment (sloria) [inline setup.py]: This change is unrelated to this PR.

**Comment by sloria** [comment]
Closing this for now; see my suggestion for splitting this into two PRs.

File: .gitignore (hunk 1 above)
Comment (brl0) [reply .gitignore]: Thanks for the info, very helpful.

File: .konchrc (hunk 2 above)
Comment (brl0) [reply .konchrc]: Understood.  It was added because it allows VS Code to recognize it as Python code since it does not have a .py extension.  Probably just trying to be unnecessarily clever.

File: .konchrc (hunk 5 above)
Comment (brl0) [reply .konchrc]: These functions were being used for troubleshooting, they were left thinking they were helpful, but definitely not needed for the fix, I will not include in new PRs.

File: CHANGELOG.rst (hunk 6 above)
Comment (brl0) [reply CHANGELOG.rst]: Understood, I couldn't decide what I should put there, but now it seems obvious.
Thanks for your patience.
